### ISSUE TITLE
Accept websocket subprotocols

### ISSUE THREAD

**Issue by hkorte** [http4k/http4k#885]
I am trying to simulate a Rails backend offering a websocket endpoint with the subprotocol "actioncable-v1-json". For this to work, the response to the upgrade request has to contain the header `Sec-WebSocket-Protocol: actioncable-v1-json`.

Unfortunately, I haven't found an official way to tell the Jetty server to accept this protocol. What works, though, is using a cloned version of the `PolyServerConfig` implementation `Jetty` using a modified `WsHandler.toJettyNegotiator()` function with one line added before [line 44 in jettyInfra.kt](https://github.com/http4k/http4k/blob/master/http4k-server/jetty/src/main/kotlin/org/http4k/server/jettyInfra.kt#L44):
```kotlin
fun WsHandler.toJettyNegotiator() = object : WebSocketNegotiator.AbstractNegotiator() {
    override fun negotiate(negotiation: WebSocketNegotiation): FrameHandler {
        negotiation.subprotocol = "actioncable-v1-json"         // <--- the added line
        val request = negotiation.request.asHttp4kRequest()!!
        return Http4kWebSocketFrameHandler(this@toJettyNegotiator(request), request)
    }
}
```

Up to now there doesn't seem to be a more elegant way to achieve this (see [slack thread](https://kotlinlang.slack.com/archives/C5AL3AKUY/p1680875996156929?thread_ts=1680875450.644349&cid=C5AL3AKUY)).

Maybe some kind of custom `SubprotocolWsHandler` could hold a `protocol` property, which is then used in the `toJettyNegotiator()` function. Any thoughts on this?

**Comment by mdrodg** [comment]
I have exactly the same issue using the graphql-ws client.  It requires that I respond with `Sec-Websocket-Protocol: graphql-transport-ws` so I was forced to do a very similar hack and clone to `WsHandler.toJettyNegotiater()`.

**Comment by daviddenton** [comment]
I've made a change to the WsResponse to include the subprotocol optionally as a field, and added `ServerFilters.SetWsSubprotocol` as a WsFilter to allow setting of this value. Currently, this is supported in the Jetty server only.

Not completely sure how to fully test this, but please report back with suggestions or PRs.

It's being released as a part of v5.4.1.0
